homeassistant/components/tesla/__init__.py

```diff
@@ -55,37 +60,72 @@
 def setup(hass, base_config):
     """Set up of Tesla component."""
     config = base_config.get(DOMAIN)
-
+    if not config:
+        return True
     email = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
     update_interval = config.get(CONF_SCAN_INTERVAL)
-    if hass.data.get(DOMAIN) is None:
+    if email in configured_instances(hass):
+        for entry in hass.config_entries.async_entries(DOMAIN):
+            if email == entry.title:
+                hass.config_entries.async_update_entry(
+                    entry,
+                    data={
+                        CONF_USERNAME: email,
+                        CONF_PASSWORD: password,
+                        CONF_SCAN_INTERVAL: update_interval,
+                    },
+                )
+                break
+    else:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={
+                    CONF_USERNAME: email,
+                    CONF_PASSWORD: password,
+                    CONF_SCAN_INTERVAL: update_interval,
+                },
+            )
+        )
+    return True
+
+
+def setup_entry(hass, config_entry):
+    """Set up Tesla as config entry."""
+
+    _LOGGER.info("Loaded teslajsonpy==%s", teslajsonpy_version)
+    if DOMAIN not in hass.data:
+        config = config_entry.data
         try:
-            hass.data[DOMAIN] = {
-                "controller": teslaAPI(email, password, update_interval),
-                "devices": defaultdict(list),
-            }
+            controller = teslaAPI(
+                config[CONF_USERNAME], config[CONF_PASSWORD], config[CONF_SCAN_INTERVAL]
+            )
+            hass.data[DOMAIN] = {"controller": controller, "devices": defaultdict(list)}
```

Comment:
You need to use `config_entry.entry_id` or else you will not support multiple config entries.

Reply:
Ok. The current component didn't support multiple accounts but I'll make this change. No idea if it'll work and no way to test it.

Reply:
How does a component check its own config_entry.entry_id in other platforms? Do I need to save that somewhere?